akka-projection-core/src/main/scala/akka/projection/internal/SourceProviderAdapter.scala

```diff
@@ -7,21 +7,23 @@ package akka.projection.internal
 import java.util.Optional
 import java.util.concurrent.CompletionStage
 import java.util.function.Supplier
-
 import scala.compat.java8.FutureConverters._
 import scala.compat.java8.OptionConverters._
 import scala.concurrent.Future
-
 import akka.NotUsed
 import akka.annotation.InternalApi
+import akka.annotation.InternalStableApi
+import akka.dispatch.ExecutionContexts
+import akka.projection.BySlicesSourceProvider
 import akka.projection.javadsl
 import akka.projection.scaladsl
 import akka.stream.scaladsl.Source
+import akka.stream.javadsl.{ Source => JSource }
 
 /**
  * INTERNAL API: Adapter from javadsl.SourceProvider to scaladsl.SourceProvider
  */
-@InternalApi private[projection] class SourceProviderAdapter[Offset, Envelope](
+@InternalStableApi private[projection] class SourceProviderAdapter[Offset, Envelope](
```

Comment:
Useful to have this back to `SourceProviderAdapter`. I had added it back and deprecated this, with the rename to `JavaSourceProviderAdapter`, so the java sample could run without a local r2dbc snapshot.